### PR TITLE
feat (Python binding): Build test environment and add test cases

### LINKED ISSUE #411: Build python binding test environment and add test cases
Need cross-platform (Windows/MacOS/Linux) testing env for Python 3.6/3.7/3.8/3.9, and add test cases for each binded method.

### PR DESCRIPTION
**Pull request by bingyanglin**
# Description of change

Added testing env and test cases for python binding in Python 3.6/3.7/3.8/3.9 in Windows/MacOS/Linux

### Test environment
- [x] `tox`
- [x] Github Action workflow (for Python 3.6/3/.7/3.8/3.9, Windows/MacOS/Ubuntu)

### AccountsSynchronizer
- [x] gap_limit
- [x] address_index
- [x] execute

### AccountManager
- [x] stop_background_sync
- [x] set_storage_password
- [x] set_stronghold_password
- [x] is_latest_address_unused
- [x] store_mnemonic   
- [x] generate_mnemonic
- [x] verify_mnemonic
- [x] create_account
- [x] remove_account
- [x] sync_accounts
- [x] internal_transfer
- [x] backup
- [x] import_accounts
- [x] get_account
- [x] get_accounts
- [x] reattach
- [x] promote
- [x] retry
- [x] get_balance_change_events
- [x] get_balance_change_event_count
- [x] get_transaction_confirmation_events
- [x] get_transaction_confirmation_event_count
- [x] get_new_transaction_events
- [x] get_new_transaction_event_count
- [x] get_reattachment_events
- [x] get_reattachment_event_count
- [x] get_broadcast_events
- [x] get_broadcast_event_count

### AccountSynchronizer
- [x] gap_limit
- [x] skip_persistence
- [x] address_index
- [x] execute

### SyncedAccount
- [x] account_handle
- [x] deposit_address
- [x] messages
- [x] addresses

### Transfer
- [x] constructor

### AccountHandle
- [x] sync
- [x] transfer
- [x] retry
- [x] promote
- [x] reattach
- [x] id
- [x] signer_type
- [x] index
- [x] alias
- [x] created_at
- [X] last_synced_at
- [X] client_options
- [X] bech32_hrp
- [X] consolidate_outputs
- [X] generate_address
- [X] get_unused_address
- [X] is_latest_address_unused
- [X] latest_address
- [X] addresses
- [X] balance
- [X] set_alias
- [X] set_client_options
- [X] message_count
- [X] list_messages
- [X] list_spent_addresses
- [X] list_unspent_addresses
- [X] get_message

### AccountInitialiser
- [X] signer_type
- [X] alias
- [X] created_at
- [X] messages
- [X] addresses
- [X] skip_persistence
- [X] initialise

### Event

- [X] on_balance_change
- [X] remove_balance_change_listener
- [X] on_new_transaction
- [X] remove_new_transaction_listener
- [X] on_confirmation_state_change
- [X] remove_confirmation_state_change_listener
- [X] on_reattachment
- [X] remove_reattachment_listener
- [X] on_broadcast
- [X] remove_broadcast_listener
- [X] on_error
- [X] remove_error_listener
- [X] on_stronghold_status_change
- [X] remove_stronghold_status_change_listener

## Links to any relevant issues

Fixes #411 

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Added test env workflow for Python 3.6/3.7/3.8/3.9 in Windows/MacOS/Linux

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by bingyanglin** [comment]
@Hribek25 please update the docs when this PR is merged, where some typos were corrected and more methods were binded.

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -164,66 +164,66 @@ jobs:
...
-      - name: Cache cargo registry
-        uses: actions/cache@v2.1.4
-        with:
-          path: ~/.cargo/registry
-          # Add date to the cache to keep it up to date
-          key: ${{ matrix.os }}-stable-cargo-registry-${{ hashFiles('**/Cargo.lock') }}-${{ env.CURRENT_DATE }}
-          # Restore from outdated cache for speed
-          restore-keys: |
-            ${{ matrix.os }}-stable-cargo-registry-${{ hashFiles('**/Cargo.lock') }}
-            ${{ matrix.os }}-stable-cargo-registry-
```
Comment (rajivshah3) [inline .github/workflows/build-and-test.yml:233]: Was the registry cache not working?

Comment (bingyanglin) [reply .github/workflows/build-and-test.yml:233]: No. It seems not working.

**Review by lucasfernog — APPROVED** [review]

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (bingyanglin) [reply .github/workflows/build-and-test.yml:233]: I added back. Not sure why the MacOS still cannot pass the testing.

**Pull request merged**